### DISCUSSION TITLE
DuckDB 1.0.0 to query S3 files on AWS EKS

### DISCUSSION THREAD

**Discussion by JRocki** [duckdb/duckdb#12472] [Q&A] [answered]
Hello,

I have upgraded to DuckDB 1.0.0 and am trying to query parquet files on S3. Locally, I am able to do this using the following code in Python:
```
import duckdb
con = duckdb.connect()
con.query(
    f"""
    INSTALL httpfs;
    LOAD httpfs;
    CREATE OR REPLACE TEMPORARY SECRET secret (
        TYPE S3,
        PROVIDER CREDENTIAL_CHAIN
    );
    """
)

df = con.query("SELECT * FROM read_parquet('s3://my-bucket/test.parquet');")
```

However, when I deploy my project to my AWS EKS cluster, I am getting a `HTTP 403 Error`. I am using a k8s service account (rather than EC2 role) to handle my credentialing. In this same deployed environment, I am able to pull the parquet files via `boto3` so the credentials are valid. I'm wondering if the issue is that the credential chain does not handle k8s service accounts? I've tried adding the REGION and/or ENDPOINT as well as adding:
```
        INSTALL aws;
        LOAD aws;
        CALL load_aws_credentials();
```

Anyone else successfully deploy this to AWS EKS?

Thanks,

**Comment by carlopi** [comment]
Have you tried to use the more low-level:
```
CREATE SECRET secret1 (
    TYPE S3,
    KEY_ID 'AKIAIOSFODNN7EXAMPLE',
    SECRET 'wJalrXUtnFEMI/K7MDENG/bPxRfiCYEXAMPLEKEY',
    REGION 'us-east-1'
);
```
syntax?

If that works (putting in the right values), then you might try getting them from environment or config, see https://duckdb.org/docs/extensions/httpfs/s3api.html#config-provider for other options.

**Reply by JRocki** [reply]
Thanks for the response! In version 0.10.0 I was able to load in my keys and connect similar to above. I can try again, however, we are moving away from creating explicit keys for deployments in favor of k8s service accounts.

**Comment by JRocki** [comment]
Adding that the following works:
```
  session = boto3.Session()
  credentials = session.get_credentials()

  con = duckdb.connect()

  con.query(
      f"""
      INSTALL httpfs;
      LOAD httpfs;
      SET s3_region='{s3_region}';
      SET s3_access_key_id='{credentials.access_key}';
      SET s3_secret_access_key='{credentials.secret_key}';
      SET s3_session_token='{credentials.token}';
      """
  )

df = con.query("SELECT * FROM read_parquet('s3://my-bucket/test.parquet');")
```
Using Python's boto3 to get the credentials from the k8s service account and than setting them in DuckDB works. Seems like the CREDENTIAL_CHAIN should work given that this works.

**Comment by samansmink** [comment, marked as answer]
I think this is the same issue as https://github.com/duckdb/duckdb_aws/issues/31?

**Reply by JRocki** [reply]
Yes I think it is actually! I'll wait and see how this [PR](https://github.com/duckdb/duckdb_aws/pull/45) goes, happy to help test. The boto3 work around is fine for now as it let's us avoid using static keys, in case that's also an issue for anyone else.

**Reply by samansmink** [reply]
Hey @JRocki if you could try this out with the latest nightly build from duckdb v1.0.0? you can install it using:

```SQL
force install aws from core_nightly
```
